frontend/appflowy_tauri/src/appflowy_app/components/_shared/EditRow/CheckListPopup.tsx

```diff
@@ -0,0 +1,179 @@
+import { CellIdentifier } from '$app/stores/effects/database/cell/cell_bd_svc';
+import { SelectOptionCellDataPB, SelectOptionColorPB, SelectOptionPB } from '@/services/backend';
+import { PopupWindow } from '$app/components/_shared/PopupWindow';
+import { ISelectOption, ISelectOptionType } from '$app_reducers/database/slice';
+import { Details2Svg } from '$app/components/_shared/svg/Details2Svg';
+import { useAppSelector } from '$app/stores/store';
+import { useCell } from '$app/components/_shared/database-hooks/useCell';
+import { CellCache } from '$app/stores/effects/database/cell/cell_cache';
+import { FieldController } from '$app/stores/effects/database/field/field_controller';
+import { EditorCheckSvg } from '$app/components/_shared/svg/EditorCheckSvg';
+import { EditorUncheckSvg } from '$app/components/_shared/svg/EditorUncheckSvg';
+import AddSvg from '$app/components/_shared/svg/AddSvg';
+import { useTranslation } from 'react-i18next';
+import { SelectOptionCellBackendService } from '$app/stores/effects/database/cell/select_option_bd_svc';
+import { useEffect, useState } from 'react';
+import { CheckListProgress } from '$app/components/_shared/CheckListProgress';
+
+export const CheckListPopup = ({
+  left,
+  top,
+  cellIdentifier,
+  cellCache,
+  fieldController,
+  openCheckListDetail,
+  onOutsideClick,
+}: {
+  left: number;
+  top: number;
+  cellIdentifier: CellIdentifier;
+  cellCache: CellCache;
+  fieldController: FieldController;
+  openCheckListDetail: (left: number, top: number, option: SelectOptionPB) => void;
+  onOutsideClick: () => void;
+}) => {
+  const databaseStore = useAppSelector((state) => state.database);
+  const { data } = useCell(cellIdentifier, cellCache, fieldController);
+  const { t } = useTranslation('');
+
+  const [allOptionsCount, setAllOptionsCount] = useState(0);
+  const [selectedOptionsCount, setSelectedOptionsCount] = useState(0);
+  const [newOptions, setNewOptions] = useState<string[]>([]);
+
+  useEffect(() => {
+    setAllOptionsCount(
+      (databaseStore.fields[cellIdentifier.fieldId]?.fieldOptions as ISelectOptionType)?.selectOptions?.length || 0
```

Comment:
```suggestion
      (databaseStore.fields[cellIdentifier.fieldId]?.fieldOptions as ISelectOptionType)?.selectOptions?.length ?? 0
```
Is it not supposed to be `??` over `||` as it's not a condition but a null-check operation?

Reply:
yep, agreed